program/src/processor/harvest_stake_rewards.rs

```diff
@@ -13,8 +21,127 @@ use crate::instruction::accounts::{Context, HarvestStakeRewardsAccounts};
 /// 2. `[w]` Destination account
 /// 3. `[s]` Stake authority
 pub fn process_harvest_stake_rewards(
-    _program_id: &Pubkey,
-    _ctx: Context<HarvestStakeRewardsAccounts>,
+    program_id: &Pubkey,
+    ctx: Context<HarvestStakeRewardsAccounts>,
 ) -> ProgramResult {
+    // Account validation.
+
+    // config
+    // - owner must be the stake program
+    // - must be initialized
+
+    require!(
+        ctx.accounts.config.owner == program_id,
+        ProgramError::InvalidAccountOwner,
+        "config"
+    );
+
+    let mut config_data = ctx.accounts.config.try_borrow_mut_data()?;
+    let config = bytemuck::try_from_bytes_mut::<Config>(&mut config_data)
+        .map_err(|_error| ProgramError::InvalidAccountData)?;
+
+    require!(
+        config.is_initialized(),
+        ProgramError::UninitializedAccount,
+        "config",
+    );
+
+    // stake
+    // - owner must be the stake program
+    // - must be initialized
+    // - derivation must match (validates the config account)
+
+    require!(
+        ctx.accounts.stake.owner == program_id,
+        ProgramError::InvalidAccountOwner,
+        "stake"
+    );
+
+    let mut stake_data = ctx.accounts.stake.try_borrow_mut_data()?;
+    let stake = bytemuck::try_from_bytes_mut::<Stake>(&mut stake_data)
+        .map_err(|_error| ProgramError::InvalidAccountData)?;
+
+    require!(
+        stake.is_initialized(),
+        ProgramError::UninitializedAccount,
+        "stake",
+    );
+
+    let (derivation, _) =
+        find_stake_pda(&stake.validator_vote, ctx.accounts.config.key, program_id);
+
+    require!(
+        ctx.accounts.stake.key == &derivation,
+        ProgramError::InvalidSeeds,
+        "stake",
+    );
+
+    // stake authority
+    // - must be a signer
+    // - must match the authority on the stake account
+
+    require!(
+        ctx.accounts.stake_authority.is_signer,
+        ProgramError::MissingRequiredSignature,
+        "stake authority",
+    );
+
+    require!(
+        ctx.accounts.stake_authority.key == &stake.authority,
+        StakeError::InvalidAuthority,
+        "stake authority",
+    );
+
+    // Determine the stake rewards.
+
+    let accumulated_rewards_per_token = config.accumulated_stake_rewards_per_token();
+    let rewards = calculate_eligible_rewards(
+        accumulated_rewards_per_token,
+        stake.last_seen_stake_rewards_per_token(),
+        stake.amount,
+    )?;
```

Comment:
Actually, considering this is using `calculate_eligible_rewards` too, we might want to keep a copy of the function in this repo rather than reusing the one from the rewards program.